lib/boom_notifier/error_info.ex

```diff
@@ -116,4 +119,30 @@ defmodule BoomNotifier.ErrorInfo do
       Enum.reduce(options, %{}, fn opt, acc ->
         Map.merge(acc, build_custom_data(conn, opt))
       end)
+
+  # Generates a unique hash key based on the error info. The timestamp and the
+  # request info is removed so we don't get different keys for the same error.
+  #
+  # The map is converted to a string using `inspect()` so we can hash it using
+  # the crc32 algorithm that was taken from the Exception Notification library
+  # for Rails
+  @spec generate_error_key(__MODULE__.t()) :: non_neg_integer()
+  def generate_error_key(error_info) do
+    error_info
+    |> Map.delete(:request)
+    |> Map.delete(:metadata)
+    |> Map.delete(:timestamp)
+    |> Map.update(:stack, nil, fn stacktrace -> List.first(stacktrace) end)
+    |> inspect()
+    |> :erlang.crc32()
+  end
+
+  @spec ensure_key(__MODULE__.t()) :: __MODULE__.t()
+  def ensure_key(%{key: nil} = error_info) do
+    Map.put(error_info, :key, generate_error_key(error_info))
+  end
+
+  def ensure_key(%{} = error_info) do
+    Map.put_new_lazy(error_info, :key, fn -> generate_error_key(error_info) end)
+  end
```

Comment:
Why not just?

```elixir
def ensure_key(error_info) do
  Map.put_new(error_info, :key, generate_error_key(error_info))
end

Reply:
just to prevent calling generate_error_key if key is present

Reply:
Make sense! I've just realized you are splitting the case where we do have a key (but is nil) vs we don't event have the key attribute ❤️